amethyst_ui/src/action_components.rs

```diff
@@ -0,0 +1,58 @@
+use amethyst_audio::SourceHandle;
+use amethyst_core::specs::prelude::{Component, DenseVecStorage};
+use amethyst_renderer::TextureHandle;
+
+/// When this component is added to a UI element with a `UiImage`
+/// it will change that image based on mouse interaction.
+/// Requires `MouseReactive`.
+pub struct OnUiActionImage {
+    /// Default image
+    pub(crate) normal_image: TextureHandle,
+    /// Image used when the mouse hovers over this element
+    pub(crate) hover_image: Option<TextureHandle>,
+    /// Image used when element is pressed
+    pub(crate) press_image: Option<TextureHandle>,
+}
+
+impl OnUiActionImage {
+    /// A constructor for this component
+    pub fn new(
+        normal_image: TextureHandle,
```

Comment:
what if you don't want an image except you have your mouse over? empty image vs Option?

Reply:
Good point, I'll add it

Reply:
A truly empty texture is pretty difficult.  The best I can do is a single pixel with alpha 0.

Reply:
Adding and removing the sprite or texture or mesh component isn't possible from a System?

Reply:
It is, it just results in kind of sloppy code because we have to account for the possibility of the component not being there when performing the updates.  The behavior flow chart gets a bit out of control.